In [49]:
# load sql extension
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [73]:
%sql mysql://admin:isba_4715@isba-dev-01.chj0o61n9xl0.us-east-1.rds.amazonaws.com/ticketmaster

## 5 Exploratory SQL Queries

## Exploratory Query #01 

*The minimum price column displays the lowest ticket price offered in each event. This query displays the maximum value of the minimum prices.*

In [74]:
%%sql
SELECT 
    show_name,
    MAX(min_price) AS  highest_min_price
FROM price
LIMIT 1;

 * mysql://admin:***@isba-dev-01.chj0o61n9xl0.us-east-1.rds.amazonaws.com/ticketmaster
1 rows affected.


[('Iron Lung, Brain Tourniquet + More Tba', 124.99)]

As we can see, the highest minimum price of $124.99 is from the show Iron Lung, Brain Tourniquet + More Tba. This data suggests that there might be a market for higher-priced tickets for different types of events/genres. 

## Exploratory Query #02

*The query shows the number of pop concerts in LA.*

In [75]:
%%sql
SELECT 
    COUNT(*) AS num_of_pop_events
FROM shows
WHERE genre = 'Pop';

 * mysql://admin:***@isba-dev-01.chj0o61n9xl0.us-east-1.rds.amazonaws.com/ticketmaster
1 rows affected.


[(60,)]

There are a total of 60 pop events in LA out of the 200 shows we have scraped. We can conclude that there is a high demand of pop shows in LA. The popularity of these shows can depend on the popularity of the artist(s), current music trends, or the consumer demographic.

## Exploratory Query #03

*This query highlights counts the number of events that do not have a minimum or maximum price.*

In [85]:
%%sql
SELECT
    COUNT(*) AS events_without_price
FROM price
WHERE min_price = 0 OR max_price = 0;
    

 * mysql://admin:***@isba-dev-01.chj0o61n9xl0.us-east-1.rds.amazonaws.com/ticketmaster
1 rows affected.


[(27,)]

It can be seen that 27 events do not a minimum or maximum price. This could be because the have not gone on sale yet so the prices are unknown. 

## Exploratory Query #04

*The max_price column on the price table shows the maximum price of each event. Query #04 displays the highest price of all the events in LA.*

In [79]:
%%sql
SELECT 
    MAX(max_price)
FROM price;

 * mysql://admin:***@isba-dev-01.chj0o61n9xl0.us-east-1.rds.amazonaws.com/ticketmaster
1 rows affected.


[(570.0,)]

The highest ticket being sold on Ticketmaster's website is $570. This indicates that there are shows in LA that are priced at a premium. This could be because the show is high in demand or it is part of a VIP package.

## Exploratory Query #05

*Query #05 displays the average minimum and maximum price of the pop and rock events in LA.*

In [80]:
%%sql
SELECT
    shows.genre,
    ROUND(AVG(min_price), 2) AS avg_min_price,
    ROUND(AVG(max_price), 2) AS avg_max_price
FROM shows
JOIN price
    ON shows.artist_id = price.artist_id
WHERE genre = 'Pop' OR genre = 'Rock'
GROUP BY genre;

 * mysql://admin:***@isba-dev-01.chj0o61n9xl0.us-east-1.rds.amazonaws.com/ticketmaster
2 rows affected.


[('Rock', 27.37, 68.88), ('Pop', 16.75, 22.13)]

There is a signifcant difference in the price range of both genres. Rock shows has the highest average maximum price of 68.88 while Pop shows only has an average maximum price of 22.13. Likewise for the average minimum price, Pop is lower than Rock with a price of 16.75. The price difference between the two genres can be an indication of the target audience and their spending capacity.

## Primary Question: 
## In terms of live music events in Los Angeles, which venues stand out as the most active and in-demand? Can we identify the venues that host the largest number of shows? ##

*Business Justfication: This query could be used to identify the venues that host the largest number of shows. Entertainment companies will be able to determine which venues are in demand in order to make strategic decisions on where to prioritize booking their events. Lastly, this query can be used to analyze the popularity ranking of the venues.*

In [81]:
%%sql 
WITH total_num_of_shows AS (
  SELECT venue, COUNT(*) AS num_shows
  FROM information
  GROUP BY venue
),
ranked_venues AS (
  SELECT venue, num_shows, RANK() OVER (ORDER BY num_shows DESC) AS `rank`
  FROM total_num_of_shows
)
SELECT ranked_venues.`rank` AS popularity_rank, ranked_venues.venue, ranked_venues.num_shows
FROM ranked_venues
ORDER BY ranked_venues.`rank` ASC;

 * mysql://admin:***@isba-dev-01.chj0o61n9xl0.us-east-1.rds.amazonaws.com/ticketmaster
25 rows affected.


[(1, 'Greek Theatre', 34),
 (2, 'The Moroccan Lounge', 32),
 (3, 'The Echo', 21),
 (4, 'The Wiltern', 20),
 (5, 'The Peppermint Club', 17),
 (5, 'The Regent Theater', 17),
 (7, 'Teragram Ballroom', 12),
 (8, 'The Belasco', 11),
 (9, 'Orpheum Theatre', 8),
 (10, 'Crypto.com Arena', 6),
 (11, 'Echoplex', 4),
 (12, 'Hollywood Bowl', 3),
 (13, 'Harvard & Stone', 2),
 (13, 'BMO Stadium', 2),
 (15, 'The Mint', 1),
 (15, 'Academy LA', 1),
 (15, 'First Street Pool and Billiard Parlor ', 1),
 (15, 'Luckman Fine Arts Complex', 1),
 (15, 'Hollywood Pantages Theatre', 1),
 (15, 'The Theatre at Ace Hotel', 1),
 (15, 'The Fonda Theatre', 1),
 (15, 'Greek Theatre-Los Angeles', 1),
 (15, 'Shrine Auditorium-CA', 1),
 (15, 'The Novo by Microsoft', 1),
 (15, 'The Ford', 1)]

As we can see, the top 5 venues are Greek Theatre, The Moroccan Lounge, The Echo, The Wiltern, and tied in 5th place, The Peppermint Club and The Regent Theatre. Greek Theatre is clearly the most popular venue with 34 shows. It is followed closely by The Moroccan Lounge with 32 shows. This indicates that these two venues may be high in demand and could potentially be a profitable option for hosting live music events.The least popular venues with 1 shows include, The Mint, Academy LA, The Ford. I recommend for entertainment companies to prioritize booking their artists' shows at the popular venues to garner more attendees and potentially increase revenue. It is best to also look into as to why some of these venues are more popular than the others. Event planners of these companies should look into the capacity of the venues as well as the location as it can impact the attendance.

## Business Question #02:
## For the venues with at least 20 shows or the top 3 venues, what is the price range? Which genre has the highest average maximum ticket prices? ##


*Business Justification: This query can be used by entertainment companies to see whether or not there is a correlation between a venue's popularity and its ticket prices. It could also be used to identify the genre with the highest average maximum ticket prices in order to develop a strategy on how to maximize their revenue potential when they are booking events at these popular venues.*

In [88]:
%%sql
WITH popular_venues AS (
  SELECT venue
  FROM information
  GROUP BY venue
  HAVING COUNT(*) >= 20
), 
venue_price_range AS (
  SELECT p.event_id, p.min_price, p.max_price, i.venue
  FROM price p
  JOIN information i ON p.event_id = i.event_id
  WHERE i.venue IN (SELECT venue FROM popular_venues)
),
genre_ticket_prices AS (
  SELECT 
    s.genre, 
    i.venue,
    ROUND(AVG(p.min_price), 2)AS avg_min_price, 
    ROUND(AVG(p.max_price), 2) AS avg_max_price
  FROM shows s
  JOIN information i ON s.show_name = i.show_name
  JOIN venue_price_range vp ON i.event_id = vp.event_id
  JOIN price p ON i.event_id = p.event_id
  GROUP BY s.genre, i.venue
),
min_max_prices AS (
  SELECT 
    venue_price_range.venue, 
    MIN(venue_price_range.min_price) AS min_price, 
    MAX(venue_price_range.max_price) AS max_price,
    genre_ticket_prices.genre,
    genre_ticket_prices.avg_min_price,
    genre_ticket_prices.avg_max_price
  FROM venue_price_range
  JOIN information i ON venue_price_range.event_id = i.event_id
  JOIN shows s ON i.show_name = s.show_name
  JOIN genre_ticket_prices ON s.genre = genre_ticket_prices.genre AND i.venue = genre_ticket_prices.venue
  GROUP BY venue_price_range.venue, genre_ticket_prices.genre
)
SELECT 
  min_max_prices.venue,
  min_max_prices.min_price,
  min_max_prices.max_price,
  min_max_prices.genre,
  min_max_prices.avg_min_price,
  min_max_prices.avg_max_price
FROM min_max_prices
ORDER BY min_max_prices.venue ASC;


 * mysql://admin:***@isba-dev-01.chj0o61n9xl0.us-east-1.rds.amazonaws.com/ticketmaster
8 rows affected.


[('Greek Theatre', 0.0, 175.0, 'Pop', 16.77, 30.68),
 ('Greek Theatre', 0.0, 179.5, 'Rock', 31.22, 100.0),
 ('The Echo', 15.0, 20.0, 'Pop', 16.69, 16.69),
 ('The Echo', 10.0, 20.0, 'Rock', 16.4, 16.4),
 ('The Moroccan Lounge', 10.0, 20.0, 'Rock', 14.42, 14.42),
 ('The Moroccan Lounge', 12.0, 25.0, 'Pop', 16.15, 16.15),
 ('The Wiltern', 25.0, 86.0, 'Pop', 31.33, 48.42),
 ('The Wiltern', 0.0, 125.0, 'Rock', 36.52, 66.68)]

It can be observed that the venues with more than 20 shows have various price ranges for their events. The highest maximum ticket price of 179.5 and average maximum price of 100.0 belongs to the rock genre shows hosted by the Greek Theatre. On the other hand, the lowest maximum ticket prices can be seen at the Moroccan Lounge and The Echo for the rock genre shows. Rock genre events also have a higher average maximum ticket prices compared to the pop genre events at The Wiltern. This could suggest that rock events have a higher demand among the audience at the in demand venues, leading to higher ticket prices. However, the average minimum prices for pop genre shows are higher than the rock genre shows at The Echo and The Wiltern. This could indicate that pop events at these venues are priced higher across the board. It is recommended for the events with lower budgets, especially for those who are looking to host rock shows, it would be best to consider booking at The Moroccan Lounge, which offers lower ticker prices for rock genre events. It is best for pop genre shows to be hosted at The Wiltern and Greek Theatre for the rock genre shows as it could be a more profitable choice. It is recommended for event planners to consider genre of their event(s) when choosing the right venue. 

## Business Question #03:
## Can we determine any trends in the frequency of shows in different seasons?

*Business Justification: This query could be used to determine the number of shows hosted during the Summer (June, July, and August) and Fall (September and October). Companies can use this to better plan their events and adjust their marketing and promotion strategies. They can also use this query to evaluate the demand for shows during different seasons.*

In [83]:
%%sql
CREATE VIEW total_shows_diff_seasons AS
SELECT
    YEAR(event_date) AS event_year,
    CASE
        WHEN MONTH(event_date) IN (6, 7, 8) THEN 'Summer'
        WHEN MONTH(event_date) IN (9, 10) THEN 'Fall'
    END AS season,
    COUNT(*) AS num_events
FROM information
GROUP BY
    event_year,
    season;

 * mysql://admin:***@isba-dev-01.chj0o61n9xl0.us-east-1.rds.amazonaws.com/ticketmaster
(MySQLdb.OperationalError) (1050, "Table 'total_shows_diff_seasons' already exists")
[SQL: CREATE VIEW total_shows_diff_seasons AS
SELECT
    YEAR(event_date) AS event_year,
    CASE
        WHEN MONTH(event_date) IN (6, 7, 8) THEN 'Summer'
        WHEN MONTH(event_date) IN (9, 10) THEN 'Fall'
    END AS season,
    COUNT(*) AS num_events
FROM information
GROUP BY
    event_year,
    season;]
(Background on this error at: http://sqlalche.me/e/14/e3q8)


In [84]:
%%sql
SELECT
    event_year,
    season,
    num_events
FROM total_shows_diff_seasons
WHERE season IN ('Summer', 'Fall')
ORDER BY num_events DESC;

 * mysql://admin:***@isba-dev-01.chj0o61n9xl0.us-east-1.rds.amazonaws.com/ticketmaster
2 rows affected.


[(2023, 'Summer', 100), (2023, 'Fall', 39)]

It can be seen that there are more shows hosted during the summer compared to fall. In 2023, there are 100 shows being hosted in the summer while only 39 are scheduled in the fall. This could indicate that summer is a more popular time for events. A recommendation for entertainment companies is to take advantage of the high show frequency during the summer by planning and scheduling their events during this season as it can lead to a larger audience and higher revenue. If they cannot book a venue during the summer, companies could consider offering discounts or bundles during the fall season to attract more attendees and increase their revenue. 